etlman/projects/tests/test_views.py

```diff
@@ -0,0 +1,94 @@
+from http import HTTPStatus
+from typing import Optional
+
+import pytest
+from django.forms import model_to_dict
+from django.test import Client
+from django.urls import reverse
+
+from etlman.projects.models import Step
+from etlman.projects.tests.factories import PipelineFactory, StepFactory
+from etlman.projects.views import MessagesEnum
+
+
+@pytest.mark.django_db
+class TestScriptView:
+
+    client = Client()
+
+    def test_edit_script_view_status_code(self):
+        response = self.client.get(reverse("projects:step_form_upsert"))
+        assert response.status_code == HTTPStatus.OK.numerator
+
+    def test_content_of_script_is_in_HTML(self):
+        # TODO: 132 right here is being discarded, and it is a work around, very hackish
```

Comment:
Could you expand this comment to say why it's needed? I'm not sure I understand.

Alternatively, it could be simpler and more explicit to define a dictionary of the data that need to be posted.

Reply:
We were manually assigning fields before, but moved to a "factory into model_to_dict" approach so there is less room for error, the thing is, for whatever reason, the factory generates a "None" for id, and this was being kinda rejected from the endpoint not being able to parse it, this needs to be improved 100%.

Reply:
Fixed it by filtering out None values before sending the data to the POST :smile: